lib/galaxy/tools/parameters/basic.py

```diff
@@ -191,10 +185,9 @@ def to_param_dict_string( self, value, other_values={} ):
         return value
 
     def validate( self, value, trans=None ):
-        if value == "" and self.optional:
-            return
-        for validator in self.validators:
-            validator.validate( value, trans )
+        if value is not '' or not self.optional:
```

Comment:
Its a revision of the above line, saying that values which are empty and optional are not validated, but optional values which are set to a non-empty value are validated.